### PR TITLE
Improve Spark test config

### PR DESCRIPTION
**Pull request by pstutz**
**EDIT:** _Initially, this PR enabled parallel test execution, but I disabled it because the improvement was mainly due to setting the number of shuffle partitions to 1. The data below is still with parallel execution, see comment below for sequential performance._

Includes changes to prevent race conditions and resource allocation conflicts.

This change reduces the running time of all CAPS tests by ~20% on my local machine:
**Master:**
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Cypher for Apache Spark ............................ SUCCESS [  5.501 s]
[INFO] Cypher for Apache Spark - Trees .................... SUCCESS [  4.578 s]
[INFO] Cypher for Apache Spark - API ...................... SUCCESS [  2.959 s]
[INFO] Cypher for Apache Spark - IR ....................... SUCCESS [  6.788 s]
[INFO] Cypher for Apache Spark - Logical Planning ......... SUCCESS [  7.532 s]
[INFO] Cypher for Apache Spark - CORA ..................... SUCCESS [  3.887 s]
[INFO] Cypher for Apache Spark - Spark .................... SUCCESS [02:37 min]
[INFO] Cypher for Apache Spark - Cypher on Scala Collections SUCCESS [  2.734 s]
[INFO] Cypher for Apache Spark - TCK ...................... SUCCESS [02:28 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:40 min
```

**With PR:**
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Cypher for Apache Spark ............................ SUCCESS [  4.564 s]
[INFO] Cypher for Apache Spark - Trees .................... SUCCESS [  3.716 s]
[INFO] Cypher for Apache Spark - API ...................... SUCCESS [  3.511 s]
[INFO] Cypher for Apache Spark - IR ....................... SUCCESS [  7.010 s]
[INFO] Cypher for Apache Spark - Logical Planning ......... SUCCESS [  7.237 s]
[INFO] Cypher for Apache Spark - CORA ..................... SUCCESS [  3.155 s]
[INFO] Cypher for Apache Spark - Spark .................... SUCCESS [01:52 min]
[INFO] Cypher for Apache Spark - Cypher on Scala Collections SUCCESS [  2.574 s]
[INFO] Cypher for Apache Spark - TCK ...................... SUCCESS [02:05 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 04:30 min
```

### CODE REVIEW

**Comment by pstutz** [comment]
Disabled parallel test execution. The improvement is mainly due to the Spark config change, and the extra 10 seconds are not worth getting test results out of order in the log.

File: caps-spark/src/test/scala/org/opencypher/caps/demo/GCDemoTest.scala [hunk 1]
```diff
@@ -34,7 +34,7 @@ import scala.collection.Bag
 class GCDemoTest extends CAPSTestSuite with SparkSessionFixture with Neo4jServerFixture with MiniDFSClusterFixture {
 
   protected override def hdfsURI: URI = new URIBuilder(super.hdfsURI).setScheme("hdfs+csv").build()
-  protected override val dfsTestGraphPath = "/csv/prod"
+  protected override def dfsTestGraphPath = "/csv/prod"
```
Comment (DarthMax) [inline caps-spark/src/test/scala/org/opencypher/caps/demo/GCDemoTest.scala]: Is that still needed without the parallel execution?

File: caps-spark/src/test/scala/org/opencypher/caps/impl/spark/CAPSSessionHDFSTest.scala [hunk 2]
```diff
@@ -31,7 +31,7 @@ import scala.collection.Bag
 
 class CAPSSessionHDFSTest extends CAPSTestSuite with MiniDFSClusterFixture with RecordMatchingTestSupport {
 
-  protected override val dfsTestGraphPath = "/csv/sn"
+  protected override def dfsTestGraphPath = "/csv/sn"
```
Comment (DarthMax) [inline caps-spark/src/test/scala/org/opencypher/caps/impl/spark/CAPSSessionHDFSTest.scala]: See above

File: caps-spark/src/test/scala/org/opencypher/caps/test/fixture/MiniDFSClusterFixture.scala [hunk 3]
```diff
@@ -19,34 +19,67 @@ import java.net.URI
...
+      cluster.getFileSystem.copyFromLocalFile(
+        new Path(getClass.getResource(path).toString),
+        new Path(path))
+      clusters += path -> cluster
+      clusterUsageCountsByPath += path -> 1
+      cluster
+    }
+  }
+
+  def releaseClusterForPath(path: String): Unit = synchronized {
```
Comment (DarthMax) [inline caps-spark/src/test/scala/org/opencypher/caps/test/fixture/MiniDFSClusterFixture.scala]: How do we make sure that the test data is cleaned after every test?

**Pull request merged**